workflow/controller/controller.go

```diff
@@ -789,6 +795,32 @@ func (wfc *WorkflowController) runWorker() {
 	}
 }
 
+func (wfc *WorkflowController) runArchiveWorker() {
+	ctx := context.TODO()
```

Comment:
This seems like it should either pass through the existing context, as the other functions do, or be identical to `runWorker` above, which also has a panic/crash handler